File: web/iniciando-com-typescript/src/tipos/produto.ts
// nome, preco, desconto (0-1), Função precoComDesconto

let produto: {
     nome: string,
     preco: number,
     desconto: number,
     precoComDesconto: () => number
}

produto = {
    nome: 'Caneta',
    preco: 15.95,
    desconto: 0.1,
    precoComDesconto() {
        return this.preco * (1 - this.desconto)
    }
}

console.log(produto.nome)
console.log(produto.preco)
console.log(produto.desconto)
console.log(produto.precoComDesconto())

File: java/formacao-java/java-classes-metodos/src/basico/Produto.java
package basico;

public class Produto {
    int id;
    String nome;
    double preco;
    double desconto;

    @Override
    public String toString() {
        return String.format("Produto: %d - %s - R$ %.2f (Desconto: %.2f%%)", this.id, this.nome, this.preco, this.desconto);
    }
}
